src/hivpy/sexual_behaviour.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+
+from . import sex_behaviour_data as sb
+
+
+class SexualBehaviourModule:
+
+    def __init__(self, **kwargs):
+        # Randomly initialise sexual behaviour group transitions
+        self.sex_behaviour_trans_male = np.random.choice(sb.sex_behaviour_trans_male_options)
+        self.sex_behaviour_trans_female = np.random.choice(sb.sex_behaviour_trans_female_options)
+        self.baseline_risk = sb.baseline_risk  # Baseline risk appears to only have one option
+        self.sex_mixing_matrix_female = np.random.choice(sb.sex_mixing_matrix_female_options)
+        self.sex_mixing_matrix_male = np.random.choice(sb.sex_mixing_matrix_male_options)
+        self.short_term_partners_male = np.random.choice(sb.short_term_partners_male_options)
+        self.short_term_partners_female = np.random.choice(sb.short_term_partners_female_options)
+
+    # Here we need to figure out how to vectorise this which is currently blocked
+    # by the gender if statement
+    def prob_transition(self, gender, age, i, j):
+        """Calculates the probability of transitioning from sexual behaviour
+        group i to group j, based on gender and age."""
+        if(gender == 1):
+            transition_matrix = self.sex_behaviour_trans_female
+            gender_index = 1
+        else:
+            transition_matrix = self.sex_behaviour_trans_male
+            gender_index = 0
+
+        age_index = min((int(age)-15)//5, 9)
+
+        risk_factor = self.baseline_risk[age_index][gender_index]
```

Comment:
I'm missing where this comes into play in the original model at the moment but I'll keep looking

Reply:
This is eq 1 on page 8 of the model details doc, but I'm not sure where it comes into the SAAS code. 
It seems to make the male version work but cause the female version to fail though, because the female matrices seem to already be normalised, so something definitely has to change but I'm not sure what yet.